prow/tide/tide.go

```diff
@@ -1150,7 +1151,21 @@ func (c *Controller) presubmitsByPull(sp *subpool) (map[int][]config.Presubmit,
 		}
 	}
 
-	for _, ps := range c.config().Presubmits[sp.org+"/"+sp.repo] {
+	unfilteredPresubmits := c.config().Presubmits[sp.org+"/"+sp.repo]
+	if c.config().InRepoConfigFor(sp.org, sp.repo).Enabled {
+		var headRefs []string
+		for _, pr := range sp.prs {
+			headRefs = append(headRefs, string(pr.HeadRefOID))
+		}
+
+		inRepoConfig, err := inrepoconfigapi.New(c.logger, c.config(), c.gc, sp.org, sp.repo, sp.sha, headRefs)
+		if err != nil {
+			return nil, fmt.Errorf("failed to get get jobs from %q: %v", inrepoconfigapi.ConfigFileName, err)
+		}
+		unfilteredPresubmits = append(unfilteredPresubmits, inRepoConfig.Presubmits...)
```

Comment:
This is a bit sticky because we can't necessarily trust the PRs, right? If I open one to remove all jobs, this logic will just have tide merge it happily because the output of my PR is a config without any jobs

Reply:
Yeah, that is the desired behaviour here. I don't see why we wouldn't trust PRs, if we get to this place we already went through the whole `verify author is member or pr has ok-to-test or pr has /lgtm /approve` chain.